infra/services/aiSearch-post.ps1

```diff
@@ -0,0 +1,181 @@
+param(
+    [string] $ResourceGroupName,
+    [string] $BaseName
+)
+
+$ErrorActionPreference = "Stop"
+
+. "$PSScriptRoot/../../eng/common/scripts/common.ps1"
+
+# Get current subscription ID
+$subscriptionId = (Get-AzContext).Subscription.Id
+
+$apiVersion = '2020-06-30'
+$token = Get-AzAccessToken -ResourceUrl https://search.azure.com -AsSecureString | Select-Object -ExpandProperty Token
+$uri = "https://$BaseName.search.windows.net"
+
+$indexDefinition = @{
+  name = "products"
+  fields = @(
+    @{ name = "chunk_id"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true; sortable = $true; key = $true; analyzer = "keyword" }
+    @{ name = "parent_id"; type = "Edm.String"; filterable = $true; retrievable = $true; stored = $true }
+    @{ name = "chunk"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true }
+    @{ name = "title"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true }
+    @{ name = "header_1"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true }
+    @{ name = "header_2"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true }
+    @{ name = "header_3"; type = "Edm.String"; searchable = $true; retrievable = $true; stored = $true }
+    @{ name = "text_vector"; type = "Collection(Edm.Single)"; searchable = $true; retrievable = $true; stored = $true; dimensions = 1536; vectorSearchProfile = "products-azureOpenAi-text-profile" }
+    @{ name = "category"; type = "Edm.String"; searchable = $true; filterable = $true; retrievable = $true; stored = $true; facetable = $true }
+  )
+}
+
+$dataSourceDefinition = @{
+  name = "products-datasource"
+  type = "azureblob"
+  credentials = @{ connectionString = "ResourceId=/subscriptions/$subscriptionId/resourceGroups/$ResourceGroupName/providers/Microsoft.Storage/storageAccounts/$BaseName;" }
+  container = @{ name = "searchdocs" }
```

Comment:
I'm not a Bicep expert, the Bicep step for creating the container says .../default/searchdocs, does that "default" mean something else? As long as the container name matches, it's all good. If you mean to have a prefix in addition to the container, there's a separate field in the data source for that.

Reply:
It's the blobServices default resource name in the storage account:

https://learn.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts/blobservices?pivots=deployment-language-bicep#microsoftstoragestorageaccountsblobservices

Reply:
Containers sit in the blob services in the storage account